core/src/main/java/org/elasticsearch/monitor/jvm/HotThreads.java

```diff
@@ -132,32 +136,23 @@ private static boolean isIdleThread(ThreadInfo threadInfo) {
     private String innerDetect() throws Exception {
         StringBuilder sb = new StringBuilder();
 
-        if (Constants.FREE_BSD) {
-            sb.append("hot_threads is not supported on FreeBSD");
-            return sb.toString();
-        }
-
-        sb.append("Hot threads at ");
-        sb.append(DATE_TIME_FORMATTER.printer().print(System.currentTimeMillis()));
-        sb.append(", interval=");
-        sb.append(interval);
-        sb.append(", busiestThreads=");
-        sb.append(busiestThreads);
-        sb.append(", ignoreIdleThreads=");
-        sb.append(ignoreIdleThreads);
-        sb.append(":\n");
-
-        ThreadMXBean threadBean = ManagementFactory.getThreadMXBean();
-        boolean enabledCpu = false;
         try {
-            if (threadBean.isThreadCpuTimeSupported()) {
-                if (!threadBean.isThreadCpuTimeEnabled()) {
-                    enabledCpu = true;
-                    threadBean.setThreadCpuTimeEnabled(true);
-                }
-            } else {
-                throw new IllegalStateException("MBean doesn't support thread CPU Time");
+            ThreadMXBean threadBean = ManagementFactory.getThreadMXBean();
+            if (threadBean.isThreadCpuTimeSupported() == false) {
+                sb.append("hot_threads is not supported on this JDK");
```

Comment:
can we just return this string directly, rather than appending to an empty stringbuilder and then returning its toString?